web_client/actions.jsx

```diff
@@ -529,14 +535,16 @@ export const submitLoginForm = (form) => promiseAction(
     let result;
     return (
       rest
-        .login(login, password)
+        .logout()
+        .then(() => rest.login(login, password))
         .then((user) => dispatch(setCurrentUser(user, user.token.token)))
         .then((user) => (result = user))
         .then(() => dispatch(closeDialog()))
         .then(() => result)
         .catch(
           ({ responseJSON: { message } }) => (
             dispatch(setDialogError('login', message))
+              .then(() => dispatch(loginAnonymousUser()))
           )
```

Comment:
It just occurred to me that we'd want to move the catch up a few calls in the chain.  We wouldn't want to try to anonymously log in if something went wrong with the dialog closing, for example.